layout/src/Layout.ts

```diff
@@ -174,13 +173,8 @@ export class Layout extends HTMLElement {
     return !!(this.context && this.context.pagesGraph);
   }
 
-  private relativeUrl(url: string): string {
-    return [
-      Array(url.split('/').length - 1)
-        .fill('..')
-        .join('/'),
-      url,
-    ].join('/');
+  private getPageUrlWithoutOrigin(page: Page): string {
+    return this.context.base + page.url;
```

Comment:
new Url(page.url,this.context.base).href ?

Reply:
please read https://github.com/divriots/webcomponents-editor/pull/4084#discussion_r813225122
and I recommened to play with `new URL()` to understand what it can and what limitations it has, it doesn't meet any expecxtations about "what will happen if", try for yourself to see what I mean
URL is nowhere the same as node's `path.join` which I think many people think it is
in this case it won't normalize any slashes and will swallow pathnames in unexpected ways, so my solution is good enough if we keep that contrac
if we break the contract, we need smth else, not `new URL`, e.g. manualt parsing and slash normalization  or import from node's `path` which I'd avoid in dockit

Reply:
Hummm your comment was about "not nice" and not "onliner" But ok. I always think using platform first object with good abstraction is preferable but if you are confident enough about this I am fine. By the way the "+" for dealing with path and url is always a source of error.

Reply:
> Hummm your comment was about "not nice" and not "onliner" But ok. I always think using platform first object with good abstraction is preferable but if you are confident enough about this I am fine. By the way the "+" for dealing with path and url is always a source of error.

And I know the URL object that's why I am asking about having a second thought :)

Reply:
I gave it many thoughts and I do know where it fits well, so no worries, I'll use it when I can and where it actually simplifies code and helps to prevent writing custom logic
if I'm wrong, feel free to send alternative examples to my code, I'm ready to discuss always :) but we might have a few more backs-and-forths on this topic